test/rsa.test.js

```diff
@@ -1,90 +1,209 @@
+        expect(
+            rsa.verifySignature(
+                body,
+                rsa.generateSignature(body, privateKey),
+                publicKey.toString(),
+            ),
+        ).to.equal(true);
     });
 
     it('should return false', () => {
-        expect(rsa.verifySignature({...body, add: 'me'}, rsa.generateSignature(body, privateKey), publicKey.toString())).to.equal(false);
+        expect(
+            rsa.verifySignature(
+                { ...body, add: 'me' },
+                rsa.generateSignature(body, privateKey),
+                publicKey.toString(),
+            ),
+        ).to.equal(false);
+    });
+});
+
+const stringifyBodyTestHelper = (fn, oldImplementation = false) => {
+    // Define the test cases
+    const tests = [
+        {
+            name: 'should succeed with basic data',
+            in: {
+                b: 'test',
+                c: [
+                    { embedded: 2 },
+                ],
+                a: { z: 'value', x: 'data' },
+            },
+            expect: '{"a": {"x": "data","z": "value"},"b": "test","c": [{"embedded": 2}]}',
+        },
+        {
+            name: 'should succeed with ascii charracters',
+            in: { test: '! \\"$#%&\'()*,+-./:;<=>?[]{}~|_0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz' },
+            // double escape to check that the charracter actually gets escaped.
+            expect: '{"test": "! \\\\\\"$#%&\'()*,+-./:;<=>?[]{}~|_0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"}',
+        },
+        {
+            name: 'should succeed with non ascii charracters',
+            in: { test: 'hello & foo € bar' },
+            // double escape to check that the charracter actually gets escaped.
+            expect: '{"test": "hello & foo \\u20ac bar"}',
+        },
+        {
+            name: 'should succeed with non ascii charracters 2',
+            in: { test: 'äöuyåÅ' },
+            // double escape to check that the charracter actually gets escaped.
+            expect: '{"test": "\\u00e4\\u00f6uy\\u00e5\\u00c5"}',
+        },
+        {
+            name: 'should succeed with no added space and fail with the old implementation',
+            in: {test: '":hello'},
+            expect: '{"test": "\\":hello"}',
+            shouldFailOnOldImplementation: true,
+        },
+        {
+            name: 'should succeed with non object value and fail with the old implementation',
+            in: 'hello',
+            expect: '"hello"',
+            shouldFailOnOldImplementation: true,
+        },
+    ];
+
+    tests.forEach(test => {
+        it(test.name, () => {
+            const out = expect(fn(test.in));
+            if(test.shouldThrow) {
+                out.to.throw();
+                return;
+            }
+            /** Uncomment this to make the test suite pass with the old implementation
```

Comment:
@hoanvancs You can uncomment this comment block to make the tests pass with the old implementation because these are the cases that the old implementation doesn't support.